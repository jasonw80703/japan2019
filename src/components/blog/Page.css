.main-page-div {
  margin: 0 7% 1% 7%;
}

.main-page-header {
  text-align: center;
}

.header-jp {
  font-size: 50%;
  color: gray;
  vertical-align: middle;
}

.header-divider {
  position: relative;
  margin-top: 0;
  border: none;
  height: 3px;
  width: 40%;
  background: gray;
  margin-bottom: 1.5%;
}
