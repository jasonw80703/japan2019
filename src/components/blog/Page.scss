@import '../../assets/styles/colours';

.main-page-div {
  margin: 0 7% 1% 7%;
}

.main-page-header {
  color: $red;
  margin-left: 5%;
  font-size: 8vw;
}

.header-jp {
  color: $gray;
  margin-left: 5%;
}

#page-jumbotron {
  background-color: $light-beige;
  margin-bottom: 5%;
  border-bottom: 3px solid white;
}

.header-dates {
  margin-left: 5%;
  color: $light-red;
}