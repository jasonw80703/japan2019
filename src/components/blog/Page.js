import React, { Component } from 'react';
import { getS3ObjectImagePath } from 'helpers/getS3ObjectPath';
import PropTypes from 'prop-types';
import ScrollBackToTopArrow from 'components/blog/ScrollBackToTopArrow';
import FetchingImagesSpinner from 'components/blog/FetchingImagesSpinner';
import './Page.css';

const Preload = require('react-preload').Preload;

export default class Page extends Component {
  render() {
    const {
      children,
      location,
      locationJp,
      cityFolder,
      imagesToLoad
    } = this.props;

    const images = imagesToLoad.map((image) => (getS3ObjectImagePath(`${cityFolder}/${image}`)));

    return (
      <Preload
        loadingIndicator={<FetchingImagesSpinner />}
        images={images}
      >
        <div className='main-page-div'>
          <h1 className='main-page-header'>{location} <span className='header-jp'>{locationJp}</span></h1>
          <hr className='header-divider' />
          {children}
          <ScrollBackToTopArrow />
        </div>
      </Preload>
    );
  }
};

Page.propTypes = {
  children: PropTypes.node,
  location: PropTypes.string.isRequired,
  locationJp: PropTypes.string.isRequired,
  cityFolder: PropTypes.string.isRequired,
  imagesToLoad: PropTypes.arrayOf(PropTypes.string).isRequired,
};

Page.defaultProps = {
  imagesToLoad: [],
}