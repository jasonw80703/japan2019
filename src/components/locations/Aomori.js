import React from 'react';
import { getS3ObjectImagePath } from 'helpers/getS3ObjectPath';
import LocationHeader from 'components/locations/LocationHeader';
import Page from 'components/blog/Page';
import Image from 'components/blog/Image';
import TextBlock from 'components/blog/TextBlock';
import TextHeader from 'components/blog/TextHeader';
import ImageSlide from 'components/blog/ImageSlide';
import { LOCATIONS } from 'constants/locations';

// Map
import map from 'assets/map/aomori.jpg';
import mapModalDetails from 'assets/map/aomori.json';

// Texts
import aomoriText from 'assets/texts/aomori.json';
import captions from 'assets/images/aomori/captions.json';

const LOCATION = LOCATIONS.AOMORI;

export default function Aomori() {
  return (
    <div>
      <LocationHeader
        location={LOCATION}
        map={map}
        mapModalDetails={mapModalDetails}
      />
      <Page location={LOCATION}>
        <TextBlock text={aomoriText.getting_there_1} source={getS3ObjectImagePath('aomori/starbucks')} location={LOCATION['en']} imageOnRight={false} />
        <TextBlock text={aomoriText.getting_there_2} source={getS3ObjectImagePath('aomori/train')} location={LOCATION['en']} />
        
        <TextHeader header={aomoriText.headers.explore} />
        <TextBlock text={aomoriText.explore_1} />
        <Image source={getS3ObjectImagePath('aomori/daikon')} size='sm' location={LOCATION['en']} caption={captions.daikon} />
        <TextBlock text={aomoriText.explore_2} />
        <Image source={getS3ObjectImagePath('aomori/water')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.explore_3} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/aspam_1'),
            getS3ObjectImagePath('aomori/aspam_2'),
            getS3ObjectImagePath('aomori/aspam_3'),
            getS3ObjectImagePath('aomori/aspam_4')
          ]}
          location={LOCATION['en']}
        />

        <TextHeader header={aomoriText.headers.museum} />
        <Image source={getS3ObjectImagePath('aomori/museum')} location={LOCATION['en']} caption={captions.museum} />
        <TextBlock text={aomoriText.museum_1} />
        <Image source={getS3ObjectImagePath('aomori/nebuta_1')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.museum_2} source={getS3ObjectImagePath('aomori/nebuta_2')} location={LOCATION['en']} />
        <TextBlock text={aomoriText.museum_3} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/nebuta_3'),
            getS3ObjectImagePath('aomori/nebuta_4'),
            getS3ObjectImagePath('aomori/nebuta_5'),
            getS3ObjectImagePath('aomori/nebuta_6'),
            getS3ObjectImagePath('aomori/nebuta_7'),
            getS3ObjectImagePath('aomori/nebuta_8')
          ]}
          location={LOCATION['en']}
        />

        <TextBlock text={aomoriText.conc} />
        <ImageSlide
          images={[
            getS3ObjectImagePath('aomori/end_1'),
            getS3ObjectImagePath('aomori/end_2'),
            getS3ObjectImagePath('aomori/end_3'),
            getS3ObjectImagePath('aomori/end_4'),
            getS3ObjectImagePath('aomori/end_5'),
            getS3ObjectImagePath('aomori/end_6')
          ]}
          location={LOCATION['en']}
        />

      </Page>
    </div>
  )
}
