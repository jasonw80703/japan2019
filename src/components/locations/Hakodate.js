import React from 'react';
import LocationHeader from './LocationHeader';
import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.HAKODATE;

export default function Hakodate() {
  return (
    <div>
      <LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} />
      <Page location={LOCATION['en']} locationJp={LOCATION['jp']}></Page>
    </div>
  )
}