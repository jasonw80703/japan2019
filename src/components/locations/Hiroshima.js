import React from 'react';
import LocationHeader from './LocationHeader';
// import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.HIROSHIMA;

export default function Hiroshima() {
  return (
    <div><LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} /></div>
  )
}