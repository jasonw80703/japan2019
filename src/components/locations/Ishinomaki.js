import React from 'react';
import LocationHeader from 'components/locations/LocationHeader';
import Page from 'components/blog/Page';
// import Image from 'components/blog/Image';
// import TextBlock from 'components/blog/TextBlock';
// import TextHeader from 'components/blog/TextHeader';
import { LOCATIONS } from 'constants/locations';

// Map
import map from 'assets/map/ishinomaki.jpg';
import mapModalDetails from 'assets/map/ishinomaki.json';

const LOCATION = LOCATIONS.ISHINOMAKI;

export default function Ishinomaki() {
  return (
    <div>
      <LocationHeader
        location={LOCATION}
        map={map}
        mapModalDetails={mapModalDetails}
      />
      <Page location={LOCATION}></Page>
    </div>
  )
}
