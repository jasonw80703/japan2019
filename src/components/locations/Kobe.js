import React from 'react';
import LocationHeader from './LocationHeader';
import Page from '../blog/Page';
// import Image from '../blog/Image';
// import TextBlock from '../blog/TextBlock';
// import TextHeader from '../blog/TextHeader';
import { LOCATIONS } from '../../constants/locations';

const LOCATION = LOCATIONS.KOBE;

export default function Kobe() {
  return (
    <div>
      <LocationHeader dates={LOCATION['dates']} location={LOCATION['en']} nextLocation={LOCATION['next_location']} />
      <Page location={LOCATION['en']} locationJp={LOCATION['jp']}></Page>
    </div>
  )
}