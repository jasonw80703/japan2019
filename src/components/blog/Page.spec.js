import React from 'react';
import renderer from 'react-test-renderer';
import Page from 'components/blog/Page';
import { LOCATIONS } from 'constants/locations';

describe('<Page />', () => {

  const defaultProps = {
    location: LOCATIONS.SAN_FRANCISCO,
  };

  it('renders as expected', () => {
    const tree = renderer
      .create(<Page {...defaultProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
