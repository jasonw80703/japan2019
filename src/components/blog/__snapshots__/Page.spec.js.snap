// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders as expected 1`] = `
Array [
  <div
    className="jumbotron jumbotron-fluid"
    id="page-jumbotron"
  >
    <div
      className="container"
    >
      <h1
        className="main-page-header"
      >
        San Francisco
      </h1>
      <h3
        className="header-jp"
      >
        サンフランシスコ
      </h3>
      <h5
        className="header-dates"
      >
        Oct 16
      </h5>
    </div>
  </div>,
  <div
    className="main-page-div"
  >
    <span
      className="fas fa-arrow-circle-up fa-3x scroll-back-btn"
      onClick={[Function]}
      style={
        Object {
          "display": "none",
          "height": 40,
        }
      }
    />
  </div>,
]
`;
